llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -3267,11 +3267,23 @@ void VPPredInstPHIRecipe::execute(VPTransformState &State) {
   // also do that packing, thereby "hoisting" the insert-element sequence.
   // Otherwise, a phi node for the scalar value is needed.
   if (State.hasVectorValue(getOperand(0))) {
-    Value *VectorValue = State.get(getOperand(0));
-    InsertElementInst *IEI = cast<InsertElementInst>(VectorValue);
-    PHINode *VPhi = State.Builder.CreatePHI(IEI->getType(), 2);
-    VPhi->addIncoming(IEI->getOperand(0), PredicatingBB); // Unmodified vector.
-    VPhi->addIncoming(IEI, PredicatedBB); // New vector with inserted element.
+    auto *VecI = cast<Instruction>(State.get(getOperand(0)));
+    assert(isa<InsertElementInst>(VecI) ||
+           isa<InsertValueInst>(VecI) &&
```

Comment:
```suggestion
    assert(isa<InsertElementInst, InsertValueInst>(VecI) &&
```

should be able to combine the isa<> I htink

Reply:
I tried this originally but there's something about it being in an assert macro that causes an error. It's not able to see the parameter pack isa overload for some reason